lib/chef/provider/package/powershell.rb

```diff
@@ -53,7 +53,9 @@ def candidate_version
         # Installs the package specified with the version passed else latest version will be installed
         def install_package(names, versions)
           names.each_with_index do |name, index|
-            powershell_out(build_powershell_package_command("Install-Package '#{name}'", versions[index]), timeout: new_resource.timeout)
+            cmd = powershell_out(build_powershell_package_command("Install-Package '#{name}'", versions[index]), timeout: new_resource.timeout)
+            next if cmd.nil?
```

Comment:
Are you trying to check for an error here? next unless cmd.error? might make more sense if so

Reply:
If I remember correctly `cmd` is `nil` if the command is successful, so this is to guard against a `NoMethodError` from the next line.

Reply:
that sounds like an enormous bug in powershell_out if its true, it should always return a mixlib-shellout object.